### DISCUSSION TITLE
Lit TanStack query adapter

### DISCUSSION THREAD

**Discussion by Gabswim** [TanStack/query#6390] [Ideas]
At my current workplace, we employ a micro-frontend application that incorporates both React and Lit in different sections. As someone primarily accustomed to working with React, I was disheartened to find a lack of a Lit adapter.

In response, I took the initiative to develop the following libraries:

- `@tanstack/lit-query`
- `@tanstack/lit-query-devtools`

Furthermore, I've crafted an adapter for:
- `@tanstack/lit-table`

My intention is to contribute these enhancements to the main repository, and I am eager to engage in discussions with a maintainer regarding the most effective strategy for integration.

I appreciate your time and commend the excellent work being done on these remarkable libraries. Thank you in advance for considering these contributions.

**Comment by Gabswim** [comment]
@TkDodo, if you're interested, I can create a CodeSandbox showcasing what it could look like. I'll be happy to create the PR in the repository once the direction is set.
🙏 Thank you

**Reply by TkDodo** [reply]
Sounds great. Please also join our discord server

**Reply by klauss194** [reply]
Hi guys - i'm very interested as well.
Kind regards
K

**Comment by crutchcorn** [comment]
Just wanting to add a small link to a POC that @TkDodo did in this space previously:

https://www.youtube.com/watch?v=lKoAU8I4ez0

This wasn't production ready and there is no work in progress pull request, so this is only meant as a rough structure for anyone wanting to take this on

**Comment by justinfagnani** [comment]
@Gabswim did you publish these libraries anywhere? Are they on GitHub?

**Reply by Gabswim** [reply]
Sorry for the long silence. Here is a really minimal version of it just to discuss the architecture: https://github.com/TanStack/query/pull/7715. This is not at all ready to be merged. One of the major points to improve, in my opinion, is how to manage the QueryClient.

CC: @crutchcorn , @TkDodo @klauss194

**Reply by Gabswim** [reply]
@justinfagnani I know you are probably pretty busy, but if you ever get a chance to give some feedback on: https://github.com/TanStack/query/pull/7715
